app/utils/formValidation.ts

```diff
@@ -0,0 +1,22 @@
+export const validateMinLength = (minLength: number, field: string) => (value: string) => {
+  return value.trim().length < minLength ? `${field} must have at least ${minLength} letters. Please try again.` : null;
+}
+
+/**
+ * Validates the format of an email address.
+ * @param {string} value - The email address to validate.
+ * @returns {string|null} An error message if the email format is invalid (local-part@domain.TLD, with optional subdomains), otherwise null.
+ */
+export const validateEmail = (value: string) => {
```

Comment:
Email validation logic sourced from this [article](https://www.scaler.com/topics/email-validation-in-javascript/).

Explanation of the Regex (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/):

^: Matches the beginning of the string.
\w+: Matches one or more word characters (letters, numbers, and underscores). This captures the local-part of the email address before the "@".
([\.-]?\w+)*: Matches zero or more repetitions of the following group:
[\.-]?: Matches an optional dot (".") or hyphen ("-"). This allows for subdomains in the local-part.
\w+: Matches one or more word characters again.
@: Matches the "@" symbol.
\w+: Matches one or more word characters. This captures the domain name before the top-level domain (TLD).
([\.-]?\w+)*: Same logic as before, allowing for subdomains in the domain name.
(\.\w{2,3})+$: Matches one or more occurrences of the following group:
\.: Matches a literal dot (".") separating the domain name from the TLD.
\w{2,3}: Matches 2 or 3 word characters. This captures the TLD (e.g., "com", "org", "net").
+: Matches one or more repetitions of the preceding group (ensures at least a 2-character TLD).
$: Matches the end of the string.

This regex ensures the email address has a valid structure with a local-part, "@" symbol, domain name, and a TLD with a length of 2 or 3 characters.

Reply:
This validation function does not account for all edge cases but good enough for our use case. Could also use a validator library if more robust validation for emails is needed.

Reply:
In your comment you're missing the * after your parentheses, but you have it in the code. Just commenting for whomever comes after trying to parse the regex comment.

I don't know if you've seen the fully realized RFC compliant regex, but it's insane: https://emailregex.com/

O'Reilly has a good article about it as well: https://www.oreilly.com/library/view/regular-expressions-cookbook/9781449327453/ch04s01.html

Reply:
@dcpavel Nice catch on the `*`, just fixed that. I also replaced the email regex with a more robust one after reading the 2 articles you shared.